Abonament.java

```diff
@@ -109,26 +109,48 @@ public static void schimbDeAbonament(String ID) {
             String line;
             while ((line = br.readLine()) != null) {
                 if (line.contains(ID)) {
-                    if (Objects.equals(abAles, "Start 100")) {
-                        String newStare = "ID: " + ID + ", GB: " + Start100.gb + ", MIN: " + Start100.min;
-                        FileHelper.replaceLine("StareaContului.txt", line, newStare);
-                    }
-                    if (Objects.equals(abAles, "Max 140")) {
-                        String newStare = "ID: " + ID + ", GB: " + Max140.gb + ", MIN: " + Max140.min;
-                        FileHelper.replaceLine("StareaContului.txt", line, newStare);
-                    }
-                    if (Objects.equals(abAles, "Max 175")) {
-                        String newStare = "ID: " + ID + ", GB: " + Max175.gb + ", MIN: " + Max175.min;
-                        FileHelper.replaceLine("StareaContului.txt", line, newStare);
-                    }
-                    if (Objects.equals(abAles, "Max 200")) {
-                        String newStare = "ID: " + ID + ", GB: " + Max200.gb + ", MIN: " + Max200.min;
-                        FileHelper.replaceLine("StareaContului.txt", line, newStare);
-                    }
-                    if (Objects.equals(abAles, "Max 290")) {
-                        String newStare = "ID: " + ID + ", GB: " + Max290.gb + ", MIN: " + Max290.min;
-                        FileHelper.replaceLine("StareaContului.txt", line, newStare);
+                    String newStare;
+                    switch (abAles) {
+                        case "Start 100":
+                            newStare = "ID: " + ID + ", GB: " + Start100.gb + ", MIN: " + Start100.min;
+                            FileHelper.replaceLine("StareaContului.txt", line, newStare);
+                            break;
+                        case "Max 140":
+                            newStare = "ID: " + ID + ", GB: " + Max140.gb + ", MIN: " + Max140.min;
+                            FileHelper.replaceLine("StareaContului.txt", line, newStare);
+                            break;
+                        case "Max 175":
+                            newStare = "ID: " + ID + ", GB: " + Max175.gb + ", MIN: " + Max175.min;
+                            FileHelper.replaceLine("StareaContului.txt", line, newStare);
+                            break;
+                        case "Max 200":
+                            newStare = "ID: " + ID + ", GB: " + Max200.gb + ", MIN: " + Max200.min;
+                            FileHelper.replaceLine("StareaContului.txt", line, newStare);
+                            break;
+                        case "Max 290":
+                            newStare = "ID: " + ID + ", GB: " + Max290.gb + ", MIN: " + Max290.min;
+                            FileHelper.replaceLine("StareaContului.txt", line, newStare);
                     }
```

Comment:
aici ai linia 

> FileHelper.replaceLine("StareaContului.txt", line, newStare);

pentru orice caz din switch, efectiv iti repeti codul fara necesitate, poti sa scoti aceasta linie inafara, dupa switch.
Astfel o sa setezi valoarea la newStare in dependenta de fiecare caz aparte, dar insasi replace-ul nu depinde de newStare, o sa o chemi la fel pentru toate cazurile